src/smolvm/images/builder.py

```
@@ -1765,3 +1836,238 @@ def _do_build(
+                output_path.parent.mkdir(parents=True, exist_ok=True)
+                output_path.write_bytes(content)
+
+            logger.info("  [1/3] Building custom Docker image...")
+            build_cmd = ["docker", "build", "-t", docker_tag]
+            if docker_platform:
+                build_cmd.extend(["--platform", docker_platform])
+            for key, value in self.build_args.items():
+                build_cmd.extend(["--build-arg", f"{key}={value}"])
+            build_cmd.append(str(tmp_path))
+
+            subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)